client/my-sites/plugins/plugins-list/test/index.jsx

```diff
@@ -0,0 +1,81 @@
+import { expect } from 'chai';
+import noop from 'lodash/noop';
+
+import useMockery from 'test/helpers/use-mockery';
+import useFakeDom from 'test/helpers/use-fake-dom';
+
+import { sites } from './sites-data';
+import i18n from 'lib/mixins/i18n';
+
+describe( 'PluginsList', () => {
+	let React, ReactInjection, TestUtils;
+	let PluginsList = {};
+	let siteListMock = {
+		getSelectedSite() {
+			return sites[0];
+		}
+	};
+	useFakeDom();
+	useMockery( mockery => {
```

Comment:
Yeah, the boilerplate code I ended up with while trying isolate the component for testing bummed me out. The `PluginsList` component is fairly high up in the hierarchy so _as is_ there not much can be done about all the child components.  

We could rebuild `PluginsList` to use inversion of control for its children but I think that's a big commitment that should be evaluated as a general approach for all of Calypso's hierarchical components. I could do an experimental spike on what that would look like but I feel that is out of scope for this particular issue.

There's also `jest`. I'm a big fan of the concept of mock everything but I've only implemented it with some toy projects. Not sure how it scales for larger projects.